vision_datasets/common/data_manifest.py

```diff
@@ -502,23 +502,27 @@ def spawn(self, num_samples, random_seed=0, instance_weights: List = None):
         Args:
             num_samples (int): size of spawned manifest. Should be larger than the current size.
             random_seed (int): Random seed to use.
-            instance_weights (list): weight of each instance to spawn.
+            instance_weights (list): weight of each instance to spawn, >= 0.
 
         Returns:
             Spawned dataset (DatasetManifest)
         """
         assert num_samples > len(self)
         if instance_weights is not None:
             assert len(instance_weights) == len(self)
+            assert all([x >= 0 for x in instance_weights])
+
             sum_weights = sum(instance_weights)
             # Distribute the number of num_samples to each image by the weights. The original image is subtracted.
-            image_multipliers = [max(0, round(w / sum_weights * num_samples - 1)) for w in instance_weights]
+            n_copies_per_sample = [max(0, round(w / sum_weights * num_samples - 1)) for w in instance_weights]
             spawned_images = []
-            for image, multiplier in zip(self.images, image_multipliers):
```

Comment:
[image] * n, will keep n copies of images. If you modify one of them later, it will change all others

Reply:
Given this, could you also fix https://github.com/microsoft/vision-datasets/blob/a5177ccfe88156bea8aef3fbde14ab4dc0d192b8/vision_datasets/common/data_manifest.py#L371 which is called by line `524`?